### PR TITLE
feat(api): add `distinct` arg to `ArrayCollect`

### PR DESCRIPTION
**Pull request by cpcloud**
This PR adds a `distinct` boolean argument to `ArrayCollect` and also makes it a subclass of `Filterable` allowing the `where` argument.

### CODE REVIEW

**Comment by cpcloud** [comment]
I'm not 100% sure whether I like the `distinct=True` API, since it looks like `DISTINCT` is a property of the column, but I can't think of another way to support distinct inputs to aggregates without reintroducing `col.distinct()`.

File: ibis/backends/pandas/execution/arrays.py [hunk 1]
```diff
@@ -110,6 +110,21 @@ def execute_array_repeat_scalar(op, data, n, **kwargs):
...
 
 
-@execute_node.register(ops.ArrayCollect, (pd.Series, SeriesGroupBy))
-def execute_array_collect(op, data, aggcontext=None, **kwargs):
-    return aggcontext.agg(data, np.array)
+@execute_node.register(ops.ArrayCollect, pd.Series, bool, (type(None), pd.Series))
+def execute_array_collect(op, data, distinct, where, aggcontext=None, **kwargs):
+    return aggcontext.agg(
+        data.loc[where] if where is not None else data,
+        np.unique if distinct else np.array,
```
Comment (jreback) [inline ibis/backends/pandas/execution/arrays.py:117]: use pd.unique as np.unique sorts

Comment (cpcloud) [reply ibis/backends/pandas/execution/arrays.py:117]: Technically the order is undefined, but happy to use `pd.unique`.

Comment (jreback) [reply ibis/backends/pandas/execution/arrays.py:117]: sure - pd.unique preserves whatever it is

Comment (cpcloud) [reply ibis/backends/pandas/execution/arrays.py]: Fixed! Appreciate the review!

**Comment by cpcloud** [comment]
I'll split this into two PRs so that we can discuss the `distinct` API separate from the `Filterable` implementation, which I believe is not controversial in any way.

**Comment by cpcloud** [comment]
Closing this out for now, don't really feel like tackling this yet and there's no pressing need.

**Pull request closed without merging**